subprojects/diagnostics/src/integTest/groovy/org/gradle/api/tasks/diagnostics/TaskReportTaskIntegrationTest.groovy

```diff
@@ -174,6 +175,25 @@ b
         TASKS_DETAILED_REPORT_TASK | true
     }
 
+    def "renders only tasks in help group running [tasks --group=build-setup]"() {
+        when:
+        succeeds tasks
+
+        then:
+        output.contains("""
+Build Setup tasks
+-----------------
+init - Initializes a new Gradle build.
+wrapper - Generates Gradle wrapper files.""")
+        !output.contains("""
+Help tasks
+----------
+""")
+
+        where:
+        tasks << [TASKS_GROUP_REPORT_TASK]
+    }
+
```

Comment:
We should probably also have a test that shows the behavior when group doesn't match any tasks.  Right now it looks a bit odd:
```
$ gradle tasks --group="foo" 

> Task :tasks

------------------------------------------------------------
All tasks runnable from root project
------------------------------------------------------------

No tasks

Rules
-----
Pattern: clean<TaskName>: Cleans the output files of a task.
Pattern: build<ConfigurationName>: Assembles the artifacts of a configuration.
Pattern: upload<ConfigurationName>: Assembles and uploads the artifacts belonging to a configuration.

To see all tasks and more detail, run gradle tasks --all

To see more detail about a task, run gradle help --task <task>
```

1. Maybe this should indicate that we're looking for tasks in group "foo".  e.g., "All tasks in group 'foo' runnable from root project".  Or maybe it should say "No tasks in group 'foo'".
2. Maybe we shouldn't show the rules if you're filtering by group? I could go either way on this.

Reply:
I like just not showing the rules.